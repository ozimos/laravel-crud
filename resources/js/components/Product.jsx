import React from 'react';
import PropTypes from 'prop-types';

const Product = ({ product }) => (
  product.title
    ? (
      <div>
        <h2> {product.title} </h2>
        <h4> {product.description} </h4>
        <h3>
          Status : {product.availability ? 'Available' : 'Out of stock'}
        </h3>
        <h3> Price : {product.price} </h3>
      </div>
    )
    : <div>  Product Doesn&#39;t exist </div>
);
Product.defaultProps = {
  product: {
    title: '',
    description: '',
    price: 0,
    availability: false
  }
};

Product.propTypes = {
  product: PropTypes.shape({
    title: PropTypes.string,
    description: PropTypes.string,
    price: PropTypes.number,
    availability: PropTypes.number,
  })
};

export default Product;
