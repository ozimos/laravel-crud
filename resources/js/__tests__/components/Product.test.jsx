/* global React:false shallow:false toJson:false */

import Product from '../../components/Product';

import { displayedProduct }
  from '../__mocks__/productsMock';

describe('Product Component', () => {
  const props = {
    product: displayedProduct
  };
  const shallowSetup = () => shallow(<Product {...props} />);

  it('renders correctly', () => {
    const wrapper = shallowSetup();
    expect(toJson(wrapper)).toMatchSnapshot();

    expect(toJson(shallow(<Product />))).toMatchSnapshot();

    const outOfStockProduct = {
      ...displayedProduct,
      availability: 0
    };
    expect(toJson(shallow(<Product product={outOfStockProduct} />)))
      .toMatchSnapshot();

  });
});
