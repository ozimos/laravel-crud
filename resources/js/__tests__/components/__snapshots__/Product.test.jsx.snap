// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product Component renders correctly 1`] = `
<div>
  <h2>
     
    est
     
  </h2>
  <h4>
     
    Dolor est eligendi sequi sapiente dicta eum.
   Et libero consequuntur non laboriosam officia.
    Inventore velit esse possimus a voluptas mollitia.
     
  </h4>
  <h3>
    Status : 
    Available
  </h3>
  <h3>
     Price : 
    73
     
  </h3>
</div>
`;

exports[`Product Component renders correctly 2`] = `
<div>
    Product Doesn't exist 
</div>
`;

exports[`Product Component renders correctly 3`] = `
<div>
  <h2>
     
    est
     
  </h2>
  <h4>
     
    Dolor est eligendi sequi sapiente dicta eum.
   Et libero consequuntur non laboriosam officia.
    Inventore velit esse possimus a voluptas mollitia.
     
  </h4>
  <h3>
    Status : 
    Out of stock
  </h3>
  <h3>
     Price : 
    73
     
  </h3>
</div>
`;
